statsmodels/distributions/discrete.py

```diff
@@ -22,3 +22,28 @@ def _pmf(self, x, mu, alpha, p):
 
 genpoisson_p = genpoisson_p_gen(name='genpoisson_p',
                                 longname='Generalized Poisson')
+
+class zipoisson_gen(rv_discrete):
+    '''A Zero Inflated Poisson distribution
+    '''
+    def _argcheck(self, mu, w):
+        return True
+
+    def _logpmf(self, x, mu, w):
+        if np.isscalar(x):
+            self._logpmf_each(x, mu, w)
+        else:
+            return [self._logpmf_each(x[i], mu[i], w[i]) for i in range(len(x))]
+
+    def _logpmf_each(self, x, mu, w):
+        if x == 0:
```

Comment:
scipy has a `lazy_where` for this case, to work vectorized with a x==0 mask instead of pointwise with xxx_each